apps/src/music/views/PatternPanel.jsx

```diff
@@ -0,0 +1,90 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import styles from './soundsPanel.module.scss';
+//import FontAwesome from '@cdo/apps/templates/FontAwesome';
+
+/*
+ * Renders a UI for designing a pattern. This is currently used within a
+ * custom Blockly Field {@link FieldSounds}
+ */
+
+const PatternPanel = ({
+  library,
+  currentValue,
+  playingPreview,
+  onSelect,
+  onPreview
+}) => {
+  const [count, setCount] = useState(0);
+
+  const group = library.groups[0];
+
+  const toggleGridPoint = (sound, tick) => {
+    console.log(sound.src, tick);
+
+    const index = currentValue.events.findIndex(
+      event => event.src === sound.src && event.tick === tick
+    );
+    if (index !== -1) {
+      // If found, delete.
+      currentValue.events.splice(index, 1);
+    } else {
+      // Not found, so add.
+      currentValue.events.push({src: sound.src, tick});
+    }
+
+    setCount(count + 1);
+  };
+
+  const getGridPointSet = (sound, tick) => {
```

Comment:
Is this function done or wip? Giving on current logic I would rather rename it to hasElement(), this way it would be more obvious on what it does (in my opinion)

Reply:
Good point.  Definitely work in progress.  I call these `events` elsewhere so will adjust the function names.